File: mirea/cources/2026-2027/aii-august/05/contest/A1.cpp
#include <bits/stdc++.h>
using namespace std;
// предподсчёт суммы всех простых Решетом Эратосфена
const int NMAX = 1 << 20;
vector<int64_t> primes;
void precalc() {
    // строим решето Эратосфена за O(n log(log(n))):
    primes.assign(NMAX+1, 1);
    primes[0] = primes[1] = 0;
    for (int i = 2; i * i <= NMAX; i++)
        if (primes[i])
            for (int j = i * i; j <= NMAX; j += i)
                primes[j] = 0;
    // считаем префиксные суммы прямо внутри решета:
    for (int i = 2; i <= NMAX; i++)
        primes[i] = (primes[i] ? i : 0) + primes[i-1];
}
void solve() {
    // читаем запросы:
    int q; cin >> q;
    vector<int> r(q);
    for (auto &it : r) cin >> it;
    // отвечаем на них:
    for (auto rr : r)
        cout << primes[rr] << ' ';
    cout << '\n';
}
main() {
    precalc(); // вызываем предподсчёт один раз
    ios::sync_with_stdio(false);
    cin.tie(0);
    int tt = 1; //cin >> tt;
    while (tt --> 0) solve();
}

File: mirea/cources/2026-2027/aii-august/05/contest/A1.py
# Время: 171 мс, Память: 37688 КБ
# предподсчёт суммы всех простых решетом Эратосфена
NMAX = 1 << 20
primes = []
def precalc():
    # строим решето Эратосфена за O(n log(log(n))):
    global primes
    primes = [1] * (NMAX + 1)
    primes[0] = primes[1] = 0
    for i in range(2, int(NMAX ** 0.5) + 1):
        if primes[i]:
            for j in range(i * i, NMAX + 1, i):
                primes[j] = 0
    # считаем префиксные суммы прямо внутри решета:
    for i in range(2, NMAX + 1):
        primes[i] = (i if primes[i] else 0) + primes[i - 1]
def solve():
    # читаем запросы:
    q = int(input())
    r = list(map(int, input().split()))
    # отвечаем на них:
    print(*[primes[rr] for rr in r])
precalc() # вызываем предподсчёт один раз
tt = 1 # tt = int(input())
while tt > 0:
    solve()
    tt -= 1